release/models/bgp/openconfig-bgp-types.yang

```diff
@@ -595,6 +601,27 @@ module openconfig-bgp-types {
                 '429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[0-1][0-9]{8}|[1-3][0-9]{9}|'  +
                 '[1-9][0-9]{1,8}|[0-9])$';
       }
+      type string {
+        // Extended Link Bandwidth Community
+        // bandwidth:<2 byte asn>:<bandwidth value in bits/sec,
+        //           optionally with kilo/mega/giga suffix>
+        // Example: link-bandwidth:20:100m
+        pattern 'link-bandwidth:'                                                 +
+                '(6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}'                        +
+                '|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{1,3}|[0-9]):'           +
+                '(429496729[0-5]|42949672[0-8][0-9]|4294967[0-1][0-9]{2}'         +
+                '|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|'      +
+                '429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[0-1][0-9]{8}|[1-3][0-9]{9}|'  +
+                '[1-9][0-9]{1,8}|[0-9])(k|m|g)?';
+        oc-ext:posix-pattern '^link-bandwidth:'                                   +
+                '(6553[0-5]|655[0-2][0-9]|65[0-4][0-9]{2}'                        +
+                '|6[0-4][0-9]{3}|[1-5][0-9]{4}|[1-9][0-9]{1,3}|[0-9])$'           +
+                '(429496729[0-5]|42949672[0-8][0-9]|4294967[0-1][0-9]{2}'         +
+                '|429496[0-6][0-9]{3}|42949[0-5][0-9]{4}|4294[0-8][0-9]{5}|'      +
+                '429[0-3][0-9]{6}|42[0-8][0-9]{7}|4[0-1][0-9]{8}|[1-3][0-9]{9}|'  +
+                '[1-9][0-9]{1,8}|[0-9])(k|m|g)?';
```

Comment:
Should the `k|m|g` be upper-case vs. lower-case? It seems to me like it would be preferable to be upper-case.

Reply:
Agreed, capital is slightly more intuitive as it's canonical for 'Kilo, Mega, Giga' vs lowercase which might get confused with 'milli' per the [SI standard units](https://en.wikipedia.org/wiki/Metric_prefix).  I allowed k and K as SI formally doesn't appear to classify 'K', only 'k'.